packages/swirl-components/src/components/swirl-app-layout/swirl-app-layout.tsx

```diff
@@ -103,17 +129,49 @@ export class SwirlAppLayout {
   }
 
   componentDidLoad() {
+    this.desktopMediaQuery.onchange = this.desktopMediaQueryHandler;
+
     queueMicrotask(() => {
+      this.isDesktop = this.desktopMediaQuery.matches;
+
+      this.restoreNavExpansionState();
       this.updateContentScrollState();
       this.updateSidebarScrollState();
       this.updateNavScrollState();
     });
   }
 
   disconnectedCallback() {
+    this.desktopMediaQuery.removeEventListener?.(
+      "change",
+      this.desktopMediaQueryHandler
+    );
+
     this.mutationObserver?.disconnect();
   }
 
+  @Listen("click", { target: "document" })
```

Comment:
**question (non-blocking):** does this auto-remove the event listener when the component gets destroyed?

Reply:
Yes. From the Stencil docs:

> The Listen() decorator is for listening to DOM events, including the ones dispatched from @Events. The event listeners are automatically added and removed when the component gets added or removed from the DOM.